nxcomp/src/ChannelEndPoint.cpp

```diff
@@ -90,13 +93,13 @@ void
 ChannelEndPoint::setSpec(const char *hostName, long port) {
   int length;
 
-  if (spec_) free(spec_);
   isUnix_ = false;
```

Comment:
I'd keep `free(spec_)` here, it fits well into the scope logic.

Reply:
that line was the reason for doing all this because it caused the double-free. If we keep it we have to NULL _spec, too. Which makes the code ugly again.

Reply:
Ah, true, so move back and `NULL`. That wouldn't really make the code ugly IMHO.